src/main/c/waf_jni.c

```diff
@@ -1111,6 +867,131 @@ static bool _cache_single_class_weak(JNIEnv *env,
     return true;
 }
 
+
+static ddwaf_builder _get_builder_checked(JNIEnv *env, jclass clazz,
+                                                   jobject builder_obj)
+{
+    UNUSED(clazz);
+
+    ddwaf_builder builder = (ddwaf_builder)(intptr_t) JNI(GetLongField, builder_obj,
+                                            _builder_ptr);
+    if (JNI(ExceptionCheck)) {
+        return NULL;
+    }
+
+    if (!builder) {
+        JNI(ThrowNew, jcls_rte, "The Builder has already been cleared");
+    }
+
+    return builder;
+}
+
+JNIEXPORT void JNICALL Java_com_datadog_ddwaf_WafBuilder_removeConfigNative(JNIEnv *env, jclass clazz,
+                                                                        jobject builder, jstring path) {
+
+    UNUSED(clazz);
+    UNUSED(env);
+    if (builder && path) {
+        ddwaf_builder_remove_config(_get_builder_checked(env, clazz, builder), JNI(GetStringUTFChars, path, NULL), JNI(GetStringLength, path));
```

Comment:
Split these into two calls. After `_get_builder_checked` is called, exception must be checked.

Reply:
Same with everything here.. Errors at each step must be controlled.